manifests/repository.pp

```diff
@@ -4,10 +4,14 @@
       include ::apt
       apt::source { 'puppetlabs':
         location   => 'http://apt.puppetlabs.com',
-        repos      => 'main',
-        key        => '4BD6EC30',
-        key_server => 'pgp.mit.edu',
+        repos      => 'main dependencies',
+        key        => {
+          id     => '47B320EB4C7C375AA9DAE1A01054B7A24BD6EC30',
+          server => 'hkp://pgp.mit.edu:80',
+        },
+        notify     => Exec['apt_update'],
```

Comment:
The notify was included with puppetlabs-apt < 2.0.0. See https://github.com/puppetlabs/puppetlabs-apt/blob/1.8.0/manifests/source.pp and https://github.com/puppetlabs/puppetlabs-apt/blob/master/manifests/source.pp

I agree that the notify is not necessary but the dependency between Class['apt::update'] and the package resource is needed for puppetlabs-apt >= 2.0.0. But maybe I should file the issue to puppetlabs-apt.